### ISSUE TITLE
Keychain claims no data exist and data exist at the same time

### ISSUE THREAD

**Issue by LukasKorba** [Electric-Coin-Company/zashi-ios#1076]
#### What?
Very specific and very critical bug happened to me today when I was working on some UI simple task. I was updating some UI in settings and then tried build in a real device. The device started in create/restore screen so I filled in my seed + BD and restore failed with unknown error. So I tried to delete Zashi and reinstall, same behaviour. Another idea was to wipe the wallet and start over. Same behaviour. So I restarted device and same behaviour. 

So I started to quickly investigate what's going on and figured out that Zashi starts on create/restore because it fails to read wallet from keychain with 
`errSecItemNotFound = -25300,    /* The specified item could not be found in the keychain. */`

Restore flow then tries to create a wallet and store it in the keychain but it responds with
`errSecDuplicateItem  = -25299,    /* The specified item already exists in the keychain. */`

So basically Zashi can't find wallet and when a user tries to restore it, it fails because the wallet is already stored.

I changed the flag of keychain and it immediately started to work and even when I reverted the flag back, I wasn't able to reproduce the issue. I hope it's result of some dev builds/dev code combination and it's not a real issue and a blocker for Zashi 1.0. However we need to test this more so we can be sure because the bug basically blocks user from using Zashi. No chance to recover.

**Comment by LukasKorba** [comment]
Just switching between the 2 flags I see the error.

this flag
`kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly` fails to read wallet while

`kSecAttrAccessibleWhenUnlockedThisDeviceOnly` continues normally.

**Comment by LukasKorba** [comment]
I also see that most tests in `WalletStorageTests.swift` fail with the `kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly` flag.

**Comment by LukasKorba** [comment]
People having same issue https://forums.developer.apple.com/forums/thread/724267

**Comment by LukasKorba** [comment]
The problem we face is about Uniqueness of queries. Please read more [here](https://developer.apple.com/forums/thread/724013)

In a nutshell, the keys used for the accessibility are defined as
```
@available(iOS 4.0, *)
public let kSecAttrAccessibleWhenUnlockedThisDeviceOnly: CFString

@available(iOS 4.0, *)
public let kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly: CFString
```

Those strings are unique for every key used, the final fingerprint for read/write operation vary depending on these + all other keys set by devs (some of the keys don't play a role in Uniqueness btw but most do).

**Comment by LukasKorba** [comment]
No longer valid
